code/controllers/subsystem/html_audio.dm

```diff
@@ -0,0 +1,274 @@
+	var/assigned = FALSE
+	for(var/i in 1 to max_channels) // Go through the channels, find the first empty slot and get in there.
+		if(isnull(channel_assignment[i]))
+			assigned = TRUE
+			channel_assignment[i] = player
+			if(requires_LOS)
+				channel_requires_LOS_at_start[i] = TRUE
+			RegisterSignal(player, COMSIG_MOVABLE_MOVED, PROC_REF(handle_player_move))
+			RegisterSignal(player, COMSIG_PARENT_QDELETING, PROC_REF(deregister_player_qdel))
+			break
+	if(!assigned)
+		CRASH("YO WE OUT OF FUCKING CHANNELS ABORT ABORT ABORT TOO MANY FUCKING THINGS GOT HTML AUDIO REEL THAT SHIT IN BRO BACK THAT ASS UP")
+	list_clear_nulls(listeners) // clients be like *poof* mid proc
+	for(var/client/listener in listeners)
+		if(!listener)
+			continue
+		update_listener_volume(listener)
+
+/datum/controller/subsystem/html_audio/proc/deregister_player_qdel(atom/movable/player)
+	SIGNAL_HANDLER
+	UnregisterSignal(player, list(COMSIG_MOVABLE_MOVED, COMSIG_PARENT_QDELETING))
+	deregister_player(player)
+
+/datum/controller/subsystem/html_audio/proc/handle_player_move(atom/movable/player, atom/old_loc)
+	SIGNAL_HANDLER
+	list_clear_nulls(listeners) // clients be like *poof* mid proc
+	for(var/client/listener in listeners)
+		if(!listener)
+			continue
+		update_listener_volume(listener)
+
+/datum/controller/subsystem/html_audio/proc/deregister_player(atom/movable/player)
+	UnregisterSignal(player, list(COMSIG_MOVABLE_MOVED, COMSIG_PARENT_QDELETING))
+	var/current_channel = channel_assignment.Find(player)
+	channel_assignment[current_channel] = null
+	active_urls[current_channel] = null
+	channel_loop_status[current_channel] = null
+	channel_tts_status[current_channel] = null
+	channel_requires_LOS_at_start[current_channel] = null
+	channel_requires_LOS_at_start_listeners[current_channel] = null
+	list_clear_nulls(listeners) // clients be like *poof* mid proc
+	for(var/client/listener in listeners)
+		if(!listener)
+			continue
+		update_listener_volume(listener)
+
+/datum/controller/subsystem/html_audio/proc/play_audio(atom/movable/player, url, blips_url = null)
+	stop_looping_audio(player)
+	list_clear_nulls(listeners) // clients be like *poof* mid proc
+	var/our_id = channel_assignment.Find(player)
+	if(!our_id)
+		CRASH("YO SOME FUCKER JUST TRIED TO PLAY AUDIO WITHOUT REGISTERING, BOO THIS FUCKER")
+	if(blips_url)
+		active_urls[our_id] = list("normal" = url, "blips" = blips_url)
+		channel_tts_status[our_id] = TRUE
+	else
+		active_urls[our_id] = url
+		channel_tts_status[our_id] = null
+	channel_requires_LOS_at_start_listeners[our_id] = list() // fresh play, thus fresh listeners who are compatible
+	var/list/hearers = list()
+	if(channel_requires_LOS_at_start[our_id])
+		hearers = get_hearers_in_view(10, player)
+		for(var/mob/mob_hearing in hearers)
+			if(mob_hearing.client)
+				channel_requires_LOS_at_start_listeners[our_id].Add(mob_hearing.client) // the linter can kiss my ass
```

Comment:
Explain

Reply:
I want to see the linter result here instead of just choosing something that doesn't happen to work *now*

Reply:
And if it is truly a linter bug I would like to document it as a comment instead of just being vague

Reply:
the comment is outdated